### PR TITLE
Show private packages in API.

### PR DESCRIPTION
**Pull request by nashby**

### CODE REVIEW

File: lib/hexpm/repository/packages.ex [hunk 1]
```diff
@@ -135,4 +135,21 @@ defmodule Hexpm.Repository.Packages do
     |> Repo.all()
     |> Enum.into(%{})
   end
+
+  def accessible_user_owned_packages(user, for_user) do
```
Comment (nashby) [inline lib/hexpm/repository/packages.ex]: I'm not really sure about the function's name though. Feel free to suggest better name :)

File: lib/hexpm/repository/packages.ex [hunk 2]
```diff
@@ -135,4 +135,21 @@ defmodule Hexpm.Repository.Packages do
...
+    organizations = Users.all_organizations(for_user)
+    organization_ids = Enum.map(organizations, & &1.id)
+
+    # Atoms sort before strings
+    sorter = fn
+               %Organization{name: "hexpm"} ->
+                 :first
+               %Organization{name: name} ->
+                 name
+             end
```
Comment (ericmj) [inline lib/hexpm/repository/packages.ex]: We can change this to `fn org -> if(org.name == "hexpm", do: first, else: org.name) end` to keep it shorter.

File: lib/hexpm/repository/packages.ex (hunk 1 above)
Comment (ericmj) [reply lib/hexpm/repository/packages.ex]: Maybe `visible_packages_for_user`? Not sure.

File: test/hexpm/web/controllers/api/user_controller_test.exs [hunk 3]
```diff
@@ -147,6 +178,36 @@ defmodule Hexpm.Web.API.UserControllerTest do
...
+
+    test "show owned packages as other user", data do
+      conn =
+        build_conn()
+        |> put_req_header("authorization", key_for(data.user3))
+        |> get("api/users/#{data.user1.username}")
+
+      assert response(conn, 200) =~ data.package1.name
+      refute response(conn, 200) =~ data.package2.name
+    end
```
Comment (ericmj) [inline test/hexpm/web/controllers/api/user_controller_test.exs]: Thank you for expanding the test suite 👏

**Comment by nashby** [comment]
@ericmj everything should be fixed now!

**Comment by wojtekmach** [comment]
Thank you!

**Pull request merged**